Bug fixes

sahilbabu opened the pull request:
- Use featured image in metadata if recipe image not set
- Warning in recipe creation screen is annoying/scary for users
- We should calculate total time not make the user do it
- Clean up ingredients/instructions after paste

sahilbabu committed:
Warning in recipe creation

sahilbabu committed:
calculate total time

sahilbabu committed:
Clean up ingredients/instructions after paste

sahilbabu committed:
Use featured image in metadata

hgezim commented on src/class.ziprecipes.php:
```diff
@@ -1841,7 +1881,67 @@ public static function zrdn_check_image_editing_support() {
+     * @return type
+     */
+    public static function recipe_table_columns($columns) {
+        array_push($columns, "feature_post_image tinyint(1) NOT NULL DEFAULT '0'");
```
You can put this in `zrdn_recipe_install`. We only use `zrdn__db_recipe_columns` hook for plugins.

hgezim replied:
Since this isn't a plugin we can put this in `zrdn_recipe_install`.

sahilbabu replied:
I got it

hgezim commented on src/class.ziprecipes.php:
```diff
@@ -1841,7 +1881,67 @@ public static function zrdn_check_image_editing_support() {
+    }
+
+    public static function zrdn_featured_image_process($post_text, $post_id) {
+        $needle_old = 'id="amd-zlrecipe-recipe-';
```
Since we have the $post_id, we can just run a SQL update command and update the recipe(s) with the post id
and this saves us from having to do this whole search for the recipe business.

hgezim commented on src/class.ziprecipes.php:
```diff
@@ -1841,7 +1881,67 @@ public static function zrdn_check_image_editing_support() {
+        $featured_img = wp_get_attachment_url(get_post_thumbnail_id($post_id));
+        if (empty($recipe->recipe_image) && $featured_img) {
+            $update['recipe_image'] = $featured_img;
+            $update['feature_post_image'] = 1;
```
`1` should be `true`.

sahilbabu replied:
please see
https://dev.mysql.com/doc/refman/5.7/en/numeric-type-overview.html

BOOL, BOOLEAN

These types are synonyms for TINYINT(1). A value of zero is considered false. Nonzero values are considered true:

hgezim replied:
I understand that, but this is in PHP and it's easier to read `true` and `1`, Sahil bhai.

The PHP boolean should get cast to an int when it's saved to MySQL.

sahilbabu replied:
Okay Bhai

hgezim commented on src/class.ziprecipes.php:
```diff
@@ -1841,7 +1881,67 @@ public static function zrdn_check_image_editing_support() {
+        return $columns;
+    }
+
+    public static function zrdn_post_featured_image($post_id, $post, $update) {
```
Second argument is the updated post, third argument is the original post. See https://codex.wordpress.org/Plugin_API/Action_Reference/post_updated

sahilbabu replied:
You are correct! I will change it 
but it works fine because we have both check

hgezim commented on src/class.ziprecipes.php:
```diff
@@ -987,6 +992,7 @@ public static function zrdn_iframe_content($post_info = null, $get_info = null)
         $ss = array();
         $iframe_title = '';
         $recipe = null;
+        $feature_post_image = 0;
```
`0` should be `false`.

sahilbabu replied:
okay

hgezim commented on src/class.ziprecipes.php:
```diff
@@ -1437,8 +1445,14 @@ public static function zrdn_insert_db($post_info) {
-        $recipe['total_time'] = $total_time;
-
+        $recipe['total_time'] = self::zrdn_calculate_total_time($post_info);
+//        if (empty($recipe['recipe_image'])) {
```
Cleanup comments, please.

sahilbabu replied:
Okay

hgezim commented on src/class.ziprecipes.php:
```diff
@@ -1454,6 +1468,32 @@ public static function zrdn_insert_db($post_info) {
+            $prep_time_total = sprintf("%02d", $prep_time_hours) . ':' . sprintf("%02d", $prep_time_minutes) . ':00';
+            $cook_time_total = sprintf("%02d", $cook_time_hours) . ':' . sprintf("%02d", $cook_time_minutes) . ':00';
+            $secs = strtotime($cook_time_total) - strtotime("00:00:00");
+            $total_time = date("H:i:s", strtotime($prep_time_total) + $secs);
```
Seconds can be 0

sahilbabu replied:
But we are not using seconds like PT1H15M

hgezim replied:
Why is it called `$secs`?

sahilbabu replied:
just trying for plural but i forgot sec are plural as well 
I will correct

hgezim replied:
Issues isn't plural but since you're saying "we are not using seconds", the value has nothing to do with second(s) at all, right? So why is `$secs` needed at all?

hgezim commented on src/class.ziprecipes.php:
```diff
@@ -1454,6 +1468,32 @@ public static function zrdn_insert_db($post_info) {
+        $prep_time_minutes = (!empty(Util::get_array_value("prep_time_minutes", $post_info))) ? Util::get_array_value("prep_time_minutes", $post_info) : NULL;
+        $cook_time_hours = (!empty(Util::get_array_value("cook_time_hours", $post_info))) ? Util::get_array_value("cook_time_hours", $post_info) : NULL;
+        $cook_time_minutes = (!empty(Util::get_array_value("cook_time_minutes", $post_info))) ? Util::get_array_value("cook_time_minutes", $post_info) : NULL;
+        if ($prep_time_hours || $prep_time_minutes || $cook_time_hours || $cook_time_minutes) {
```
This check doesn't look like is needed.

hgezim commented on src/class.ziprecipes.php:
```diff
@@ -1454,6 +1468,32 @@ public static function zrdn_insert_db($post_info) {
 
+    public static function zrdn_calculate_total_time($post_info) {
+        $total_time = NULL;
+        $prep_time_hours = (!empty(Util::get_array_value("prep_time_hours", $post_info))) ? Util::get_array_value("prep_time_hours", $post_info) : NULL;
```
This should suffice:
`$prep_time_hours = Util::get_array_value("prep_time_hours", $post_info)`
Same for the ones below as well since `get_array_value` returns null and check if value exists with `isset` which works for arrays as well.

sahilbabu replied:
you are correct!

hgezim commented on src/class.ziprecipes.php:
```diff
@@ -1454,6 +1468,32 @@ public static function zrdn_insert_db($post_info) {
+            $secs = strtotime($cook_time_total) - strtotime("00:00:00");
+            $total_time = date("H:i:s", strtotime($prep_time_total) + $secs);
+            $time = explode(':', $total_time);
+            return 'PT' . (int) $time[0] . 'H' . (int) $time[1] . 'M';
```
We do we convert to int and then concat to string?

sahilbabu replied:
It's funny!
you are correct!
I am remembering it why i did 
anyways i will correct it

sahilbabu replied:
Now i find the answer of your question 
why i did (int) ... 
01,02,03 ..  convert to 1,2,3 ...
we need to remove first zero in format time.

hgezim commented on src/class.ziprecipes.php:
```diff
@@ -1454,6 +1468,32 @@ public static function zrdn_insert_db($post_info) {
+        if ($prep_time_hours || $prep_time_minutes || $cook_time_hours || $cook_time_minutes) {
+            $prep_time_total = sprintf("%02d", $prep_time_hours) . ':' . sprintf("%02d", $prep_time_minutes) . ':00';
+            $cook_time_total = sprintf("%02d", $cook_time_hours) . ':' . sprintf("%02d", $cook_time_minutes) . ':00';
+            $secs = strtotime($cook_time_total) - strtotime("00:00:00");
```
Not sure what's the point of this. Why not add cook time minutes, prep time minutes and likewise for hours?

hgezim replied:
Sorry, we're having this discussion in two places. You can just comment in one place.

hgezim commented on src/class.ziprecipes.php:
```diff
@@ -1454,6 +1468,32 @@ public static function zrdn_insert_db($post_info) {
+        return $total_time;
+    }
+
+    public static function zrdn_get_post_featured_image($post_info) {
```
No longer used. Remove.

sahilbabu replied:
Okay

hgezim commented on src/views/create-update-recipe.twig:
```diff
@@ -99,6 +98,22 @@
                 $recipeSummaryTextarea.data('caption', 'false');
             }
 
+            $(".clean-on-paste").bind('paste', function (e) {
```
Use `on` not `bind`.

hgezim commented on src/views/create-update-recipe.twig:
```diff
@@ -99,6 +98,22 @@
+            $(".clean-on-paste").bind('paste', function (e) {
+                var elem = $(this);
+                setTimeout(function () {                    
+                    var lines = elem.val().split(/\n/);      
```
`split` argument shouldn't need to be a regex, right?
I wonder if it'll work on windows as well?

sahilbabu replied:
we need here regex 
see 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/split

hgezim commented on src/views/create-update-recipe.twig:
```diff
@@ -99,6 +98,22 @@
+                        texts.push($.trim(lines[i]));    
+                      }    
+                    }   
+                    //var n = texts.toString().split(",").join("\n");  // it will break at comma  
```
cleanpu

hgezim replied:
p

hgezim commented on src/views/create-update-recipe.twig:
```diff
@@ -99,6 +98,22 @@
 
+            $(".clean-on-paste").bind('paste', function (e) {
+                var elem = $(this);
+                setTimeout(function () {                    
```
Add comments explaining what's happening in this function.

hgezim replied:
Also why setTimeout is needed.

hgezim commented on src/views/create-update-recipe.twig:
```diff
@@ -297,17 +316,6 @@
-                                    <input class="zrdn-input zrdn-is-small" type='number' min="0" id="total_minutes" name='total_time_minutes' value='{{ total_time_minutes }}' />minutes
-                                </div>
-                            </div>
-                        </div>
```
We can also removed `total_time` column from `zrdn_recipe_install` function since we're always calculating it.

sahilbabu replied:
But it will save front end calculation time and speed up page 
anyways as you suggest i will do that

hgezim replied:
I don't mind about the calculation cost but people might have old total time values in there so we'll leave the column stay for now.

sahilbabu replied:
Done!
Total will calculate when recipe view called means on run time

hgezim reviewed (changes requested):
Some cleaning up to do, Sahil.

sahilbabu committed:
optimized and clean code

sahilbabu committed:
Calculate Total time in run time

When recipe view called then we calculate total time

sahilbabu committed:
Merge branch 'master' into bug_fixes

hgezim committed:
Moved ob_start, cleanup, renamed feature to featured

hgezim committed:
Cleaned up time related stuff

hgezim committed:
Don't crash when plugin file doesn't exist

hgezim committed:
Comments and log

hgezim committed:
Bunch of image crap, etc

- not using image_css
- removed recipe_image since it was redundant
- renamed column featured_post_image to is_featured_post_image
- fixed logic for amp and image setting with is_featured_post_image

hgezim committed:
Fixed: JSON-LD showing null total time

hgezim committed:
Fixed issues with featured image changes

- changes to featured image were not taking place
- adding recipe image wasn't getting saved

blah

hgezim committed:
Using class name to hide image
